# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: mozglue/baseprofiler/public/BlocksRingBuffer.h
# Commit: 524f84e0fc78
# Full Hash: 524f84e0fc78eb8689dc6782f16f5bf71e4e5edf
# Author: Gerald Squelart <gsquelart@mozilla.com>
# Date: 2019-09-18 09:56:40
# Regressor Bug: 1576551
# File Overlap Count: 2
# Description:
#   Bug 1576551 - Use BlocksRingBuffer in ProfileBuffer - r=gregtatum
#   
#   This just replaces `ProfileBuffer`'s self-managed circular buffer with a
#   `BlocksRingBuffer`.
#   
# ==============================================================================

diff -r af093f06fb74 -r 524f84e0fc78 mozglue/baseprofiler/public/BlocksRingBuffer.h
--- a/mozglue/baseprofiler/public/BlocksRingBuffer.h	Wed Sep 18 01:19:10 2019 +0000
+++ b/mozglue/baseprofiler/public/BlocksRingBuffer.h	Wed Sep 18 01:19:12 2019 +0000
@@ -84,12 +84,13 @@
   // Near-infinite index type, not expecting overflow.
   using Index = uint64_t;
 
+ public:
   // Using ModuloBuffer as underlying circular byte buffer.
   using Buffer = ModuloBuffer<uint32_t, Index>;
+  using Byte = Buffer::Byte;
   using BufferWriter = Buffer::Writer;
   using BufferReader = Buffer::Reader;
 
- public:
   // Length type for total buffer (as PowerOfTwo<Length>) and each entry.
   using Length = uint32_t;
 
@@ -159,6 +160,13 @@
       return mBlockIndex >= aRhs.mBlockIndex;
     }
 
+    // Temporary escape hatches to let legacy code access block indices.
+    // TODO: Remove this when legacy code has been modernized.
+    uint64_t ConvertToU64() const { return uint64_t(mBlockIndex); }
+    static BlockIndex ConvertFromU64(uint64_t aIndex) {
+      return BlockIndex(Index(aIndex));
+    }
+
    private:
     // Only BlocksRingBuffer internal functions and serializers can convert
     // between `BlockIndex` and `Index`.